static/js/util/profile_edit.js

```diff
@@ -334,72 +300,94 @@ export function boundMonthYearField(keySet, label) {
     let formattedDate = _.get(profile, keySet);
 
     if (formattedDate !== undefined && formattedDate !== null) {
-      return moment(formattedDate, DATE_FORMAT);
+      return moment(formattedDate, ISO_8601_FORMAT);
     }
     return null;
   };
 
-  // Get a tuple { month, year } which contains the values being edited in the textbox
+  // Get a tuple { date, month, year } which contains the values being edited in the textbox
   // values may be strings or numbers. Otherwise return empty object.
-  let getMonthYear = () => {
-    let monthYear = _.get(profile, editKeySet, {});
+  let getTuple = () => {
```

Comment:
if this is just supposed to return an object with day/month/year as keys, i think this should be called `getDateObject`/`getDateJSON`/`getSerializedDate` and update the below references to `tuple` accordingly. `getTuple` isn't very descriptive